pkg/controller/storage/mem/storage_binding.go

```diff
@@ -38,35 +38,62 @@ func newMemStorageBinding() *memStorageBinding {
 }
 
 func (m *memStorageBinding) List() ([]*servicecatalog.Binding, error) {
-	ret := make([]*servicecatalog.Binding, len(m.bindings))
+	copy := make([]*servicecatalog.Binding, len(m.bindings))
 	i := 0
-	for _, val := range m.bindings {
-		ret[i] = val
+	for _, bin := range m.bindings {
+		copy[i] = &servicecatalog.Binding{}
+		if err := deepCopy(bin, copy[i]); err != nil {
+			return nil, err
+		}
 		i++
 	}
-	return ret, nil
+	return copy, nil
 }
+
 func (m *memStorageBinding) Get(name string) (*servicecatalog.Binding, error) {
-	ret, ok := m.bindings[name]
+	binding, ok := m.bindings[name]
 	if !ok {
-		return nil, errNoSuchBinding
+		return nil, errNoSuchInstance
+	}
+	copy := &servicecatalog.Binding{}
+	if err := deepCopy(binding, &copy); err != nil {
+		return nil, err
 	}
-	return ret, nil
+	return copy, nil
 }
+
 func (m *memStorageBinding) Create(bin *servicecatalog.Binding) (*servicecatalog.Binding, error) {
 	if _, err := m.Get(bin.Name); err == nil {
 		return nil, errBindingAlreadyExists
 	}
-	m.bindings[bin.Name] = bin
-	return bin, nil
+	copy1 := &servicecatalog.Binding{}
+	if err := deepCopy(bin, copy1); err != nil {
+		return nil, err
+	}
+	copy2 := &servicecatalog.Binding{}
+	if err := deepCopy(bin, copy2); err != nil {
+		return nil, err
+	}
+	m.bindings[bin.Name] = copy1
```

Comment:
:man_shrugging: It somehow felt like the right thing to do. I think I would generally (i.e. not _just_ in k8s) expect that whenever I invoke some kind of update operation against a data store, the output of that is the _updated_ object, as stored, and not simply the input being echoed back to me. Imagine if optimistic locking were implemented on these objects (which k8s _does_ do, by the way), the output should have an incremented version number. I think this is my roundabout way of saying I don't think we want to return a pointer to the input as the output.